components/org.wso2.carbon.identity.application.authenticator.fido/src/main/java/org/wso2/carbon/identity/application/authenticator/fido/FIDOAuthenticator.java

```diff
@@ -119,18 +119,27 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
         U2FService u2FService = U2FService.getInstance();
         AuthenticatedUser user = null;
         try {
-            //authentication page's URL.
+            // Authentication page's URL.
             String loginPage;
-            loginPage = context.getAuthenticatorProperties().get(IdentityApplicationConstants.Authenticator.FIDO
-                    .FIDO_AUTH);
-            if (StringUtils.isBlank(loginPage)){
-                loginPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL().replace("login.do",
-                        "fido-auth.jsp");
+            if (StringUtils.isNotBlank(getAuthenticatorConfig().getParameterMap()
+                    .get(FIDOAuthenticatorConstants.FIDO_AUTH))) {
+                loginPage = getAuthenticatorConfig().getParameterMap().get(FIDOAuthenticatorConstants.FIDO_AUTH);
+            } else {
+                loginPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL()
+                        .replace(FIDOAuthenticatorConstants.URI_LOGIN, FIDOAuthenticatorConstants.URI_FIDO_LOGIN);
             }
-            //username from basic authenticator.
+
+            // Username from basic authenticator.
             user = getUsername(context);
-            //origin as appID eg.: http://example.com:8080
+
+            // Retrieving AppID
+            // Origin as appID eg: https://example.com:8080
             String appID = FIDOUtil.getOrigin(request);
+            if (StringUtils.isNotBlank(getAuthenticatorConfig().getParameterMap()
+                    .get(FIDOAuthenticatorConstants.SERVER_URL))) {
+                appID = getAuthenticatorConfig().getParameterMap().get(FIDOAuthenticatorConstants.SERVER_URL);
```

Comment:
Shall we name the property named as the appID?

Reply:
Changed and updated the PR.